expo/tests/gbe/test_assign_volunteer.py

```diff
@@ -0,0 +1,275 @@
+import gbe.models as conf
+import nose.tools as nt
+from unittest import TestCase
+from django.test.client import RequestFactory
+from django.test import Client
+from django.core.exceptions import PermissionDenied
+from datetime import datetime, date, time
+import pytz
+import re
+from gbe.views import assign_volunteer
+from django.contrib.auth.models import Group
+from tests.factories.gbe_factories import (
+    PersonaFactory,
+    ProfileFactory,
+    VolunteerFactory,
+    ConferenceFactory,
+    GenericEventFactory,
+    UserFactory,
+    VolunteerWindowFactory,
+)
+from tests.functions.gbe_functions import login_as
+from scheduler.models import Event as sEvent
+from scheduler.models import (
+    Worker,
+    ResourceAllocation,
+    EventContainer
+)
+
+
+class TestAssignVolunteer(TestCase):
+    '''Tests for review_volunteer view'''
+
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.client = Client()
+        self.performer = PersonaFactory.create()
+        self.privileged_profile = ProfileFactory.create()
+        self.privileged_user = self.privileged_profile.user_object
+        group, nil = Group.objects.get_or_create(name='Volunteer Coordinator')
+        self.privileged_user.groups.add(group)
+
+    @nt.raises(PermissionDenied)
+    def test_assign_volunteers_bad_user(self):
+        ''' user does not have Volunteer Coordinator, permission is denied'''
+        volunteer = VolunteerFactory.create()
+        request = self.factory.get('volunteer/assign/%d' % volunteer.pk)
+        request.user = ProfileFactory.create().user_object
+        request.session = {'cms_admin_site': 1}
+        login_as(request.user, self)
+        response = assign_volunteer(request, volunteer.pk)
+
+    @nt.raises(PermissionDenied)
+    def test_assign_volunteers_no_profile(self):
+        ''' user does not have a profile, permission is denied'''
+        volunteer = VolunteerFactory.create()
+        request = self.factory.get('volunteer/assign/%d' % volunteer.pk)
+        request.user = UserFactory.create()
+        request.session = {'cms_admin_site': 1}
+        login_as(request.user, self)
+        response = assign_volunteer(request, volunteer.pk)
+
+    def test_assign_volunteer_no_events(self):
+        ''' assign a volunteer when there are no opportunities'''
+        volunteer = VolunteerFactory.create()
+        request = self.factory.get('volunteer/assign/%d' % volunteer.pk)
+        request.user = self.privileged_user
+        request.session = {'cms_admin_site': 1}
+        login_as(request.user, self)
+        response = assign_volunteer(request, volunteer.pk)
+        nt.assert_equal(response.status_code, 200)
+        nt.assert_true('Assign Volunteer to Opportunities' in response.content)
+
+    def test_assign_volunteer_show_current_events(self):
```

Comment:
I have trouble agreeing to "one assert per test" - I would have said "one set of conditions produces one definable result" and the number asserts should scope to what the defintion of the correct result is.

I'm definitely willing to agree that this is testing multiple conditions with different results and should therefore be broken down, to something like:
- if the volunteer is booked, the event is loaded as checked
- if the volunteer has an available window, they are shown as available
- if the volunteer is unavailable... 
- past conference events are not shown
- not-past conf. events are shown

etc... but that well may be more than one assertion, and keeping them as multiple different assertions gives me more information about what when wrong when a unit test fails.

I understand about the serial process problem, but trying a "1 assertion per test" seems to me like a lot of unnecessary repetitive setup.  Certainly it can be done well (a function call to do it) but I'm not sure that fully parellelizing in this direction is something I'd want to enfore or stick to.